File: src/main/java/edu/austral/starship/base/framework/ImageLoader.java
package edu.austral.starship.base.framework;

import processing.core.PApplet;
import processing.core.PImage;

public class ImageLoader {
    private final PApplet applet;

    public ImageLoader(PApplet applet) {this.applet = applet;}

    public PImage load(String fileName) { return applet.loadImage(fileName); }

    public PImage load(String fileName, String extension) { return applet.loadImage(fileName, extension); }
}

File: src/main/scala/edu/austral/starship/scala/base/framework/ImageLoader.scala
package edu.austral.starship.scala.base.framework

import processing.core.{PApplet, PImage}

case class ImageLoader(applet: PApplet) {
  def load(fileName: String): PImage = applet.loadImage(fileName)

  def load(fileName: String, extension: String): PImage = applet.loadImage(fileName, extension)
}
